tool/lib/commands/serve.dart

```diff
@@ -163,6 +176,10 @@ class ServeCommand extends Command {
     await processManager.runProcess(
       CliCommand.dart(
         [
+          if (debugServer) ...[
+            'run',
+            '--observe',
```

Comment:
This will fail if another process is already bound to port 8181. Do you want to provide a way to override the port or just bind to a random available port?

Reply:
binding to random available port is fine.